timemachine/fe/free_energy.py

```diff
@@ -309,6 +309,24 @@ def extract_ligand_trajectories_by_replica(self):
         return self.extract_trajectories_by_replica(ligand_idxs)
 
 
+def trajectories_by_replica_to_by_state(
+    trajectory_by_iter_by_replica: NDArray,
+    replica_idx_by_state_by_iter: Sequence[Sequence[ReplicaIdx]],
+) -> NDArray:
+    """Utility function to convert the output of `extract_trajectories_by_replica` from (replica, iters, ...) to
+    (state, iters, ...). This is useful for evaluating the trajectories of states.
+    """
+    assert len(trajectory_by_iter_by_replica.shape) == 4
+    replica_idx_by_iter_by_state = np.asarray(replica_idx_by_state_by_iter).T
+    assert replica_idx_by_iter_by_state.shape == trajectory_by_iter_by_replica.shape[:2]
+
+    samples_by_iter_by_state = np.take_along_axis(
```

Comment:
nit: naming consistency: samples vs. trajectory

Reply:
Fixed in https://github.com/proteneer/timemachine/pull/1350/commits/139d898892e3721b159690c60d30451640afeb57